[Security Solution][Notes] - add telemetry

PhilippeOberti opened the pull request:
## Summary

This PR adds telemetry for the new notes system in Security Solution. 2 things are getting logged:
- `openNoteInExpandableFlyoutClickedEvent`: logs when the expandable flyout is being opened at the `Notes` tab, from the alerts table on the alerts page and cases page, from timeline and from the events table on the Explore pages
- `addNoteFromExpandableFlyoutClickedEvent: logs when a new note is added. It has a boolean to indicate when the note was related to a saved timeline

### TODO

- [x] wait for CI to finish and fix some most likely broken unit tests
- [ ] create a dashboard to visualize the telemetry data (see [this PR](https://github.com/elastic/kibana/pull/178683) for example)

https://github.com/elastic/security-team/issues/9375

PhilippeOberti committed:
[Security Solution][Notes] - add telemetry

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/219151)
* Commit: 8c58d891fc3739d0b2a566d7586a59fed34e8e1d

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/219151#019073e5-a390-4646-8ba5-91470591c291)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/219151","number":219151,"commit":"8c58d891fc3739d0b2a566d7586a59fed34e8e1d"}],"number":219151}
buildkite-pr-comment-->

christineweng commented on x-pack/plugins/security_solution/public/notes/telemetry/types.ts:
```diff
@@ -0,0 +1,31 @@
+ * 2.0.
+ */
+
+import type { RootSchema } from '@kbn/ebt';
```
why is it pulling from `@kbn/ebt` instead of `@kbn/core/public`?

PhilippeOberti replied:
not sure, I let Webstorm do the import on its own and didn't pay attention... Reverted!

PhilippeOberti replied:
removed in my last commit

christineweng commented on x-pack/plugins/security_solution/tsconfig.json:
```diff
@@ -210,5 +210,6 @@
     "@kbn/core-i18n-browser",
     "@kbn/core-theme-browser",
     "@kbn/integration-assistant-plugin",
+    "@kbn/ebt",
```
see comment above, I don't think we need this

christineweng replied:
@PhilippeOberti hmm.. this file is still showing up for some reason

christineweng commented on x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/data_table/index.tsx:
```diff
@@ -190,6 +191,10 @@ export const TimelineDataTableComponent: React.FC<DataTableProps> = memo(
               },
             },
           });
+          telemetry.reportDetailsFlyoutOpened({
```
thanks for adding telemetry here!

christineweng commented on x-pack/plugins/security_solution/public/timelines/components/timeline/tabs/query/index.tsx:
```diff
@@ -239,13 +239,21 @@ export const QueryTabContentComponent: React.FC<Props> = ({
+        });
+        telemetry.reportDetailsFlyoutOpened({
+          location: timelineId,
+          panel: 'right-left',
```
I think the `panel` type is one of right, left and preview. not sure if this will go through

PhilippeOberti replied:
yes you are correct, I had originally added that new `right-left` entry as an option but reverted the change and forgot to undo a few places I was using it... I decided to go with just `left`, while it does not completely represents what's happening I feel like it's the closest.
Thoughts?

christineweng commented on x-pack/plugins/security_solution/public/timelines/components/timeline/tabs/pinned/index.tsx:
```diff
@@ -20,6 +20,7 @@ import {
   DocumentDetailsRightPanelKey,
 } from '../../../../../flyout/document_details/shared/constants/panel_keys';
 import type { ControlColumnProps } from '../../../../../../common/types';
+import {} from '../../../../../../common/types';
```
is this line needed?

PhilippeOberti replied:
nope, removed!

christineweng reviewed (commented):
Thanks for adding telemetry! Left a couple minor comments

PhilippeOberti committed:
PR comments + fix tests + move files to telemetry folder

PhilippeOberti commented:
@christineweng thanks for the review! I've fixed all the points you raised, and also moved a few files to the `telemetry` folder to be with all the other telemetry related files. I'm not sure how I missed that in the first place...

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/219363)
* Commit: 3808c6c14de0995c128d0bcca0b6c48bc26c30ea

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/219363#0190787c-10ec-4d8f-a4e7-afe3e843087f)

### History
* :broken_heart: [Build #219151](https://buildkite.com/elastic/kibana-pull-request/builds/219151) failed 8c58d891fc3739d0b2a566d7586a59fed34e8e1d

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/219363","number":219363,"commit":"3808c6c14de0995c128d0bcca0b6c48bc26c30ea"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/219151","number":219151,"commit":"8c58d891fc3739d0b2a566d7586a59fed34e8e1d"}],"number":219363}
buildkite-pr-comment-->

christineweng reviewed (approved):
Just had a comment about that ebt import, everything else looks good!

PhilippeOberti committed:
undo tsconfig.json changes

PhilippeOberti committed:
Merge branch 'main' into notes-telemetry
